pru_firmware/src/main_pru0.c

```diff
@@ -0,0 +1,162 @@
+#include <stdint.h>
+#include <pru_cfg.h>
+#include <pru_ctrl.h>
+#include "resource_table_pru1.h"
+
+
+// Register map headers
+#include <sys_tscAdcSs.h>  // ADC - in /usr/lib/ti/pru-software-support-package/include/am335x/sys_tscAdcSs.h
+#include "cm_wkup.h"
+#include "gpio.h"
+
+
+// Shared memory definitions
+#define ENCODER_MEM_OFFSET  16
```

Comment:
I remember someone commenting on the arbitrary definition of this, maybe comment is fine, but otherwise lgtm.

Reply:
added more comments